src/Bicep.Core/Emit/ExpressionEmitter.cs

```diff
@@ -174,6 +174,25 @@ public void EmitLanguageExpression(SyntaxBase syntax)
 
         public void EmitCopyObject(string? name, ForSyntax syntax, SyntaxBase? input, string? copyIndexOverride = null, long? batchSize = null)
         {
+            // local function
+            static bool CanEmitAsInputDirectly(SyntaxBase input)
```

Comment:
>CanEmitAsInputDirectly [](start = 24, length = 22)

I have an internal task to fix this in the runtime, so the workarounds aren't needed anymore.